### PR TITLE
#5380 use Orion annotation model instead 'showProblems' method

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
use Orion annotation model instead 'showProblems' method, which more wright way to add annotations to Orion editor

### What issues does this PR fix or reference?
#5380

### Changelog
Using Orion annotation model instead 'showProblems' method to add annotations

### Release Notes
Now we are using Orion annotation model instead 'showProblems' method to add annotations

### CODE REVIEW

**Comment by slemeur** [comment]
How this PR is affecting the user?

You are missing the information for the change log / release note.

**Comment by evidolob** [comment]
Non, its bug fix

**Comment by benoitf** [comment]
@evidolob AFAIK bugfix requires a changelog entry

File: ide/che-core-orion-editor/src/main/java/org/eclipse/che/ide/editor/orion/client/OrionEditorWidget.java [hunk 1]
```diff
@@ -188,6 +194,7 @@ public OrionEditorWidget(final ModuleHolder moduleHolder,
         initWidget(UIBINDER.createAndBindUi(this));
 
         this.uiUtilsOverlay = moduleHolder.getModule("UiUtils");
+        this.annotationType = moduleHolder.getModule("OrionAnnotations").<OrionAnnotationsOverlay>cast().getAnnotationType();
```
Comment (vparfonov) [inline ide/che-core-orion-editor/src/main/java/org/eclipse/che/ide/editor/orion/client/OrionEditorWidget.java:198]: maybe 
`this.annotationType = ((OrionAnnotationsOverlay)moduleHolder.getModule("OrionAnnotations")).getAnnotationType(); `
more readable?

**Comment by vparfonov** [comment]
@evidolob  please don't name branch with leading  '#'

File: ide/che-core-orion-editor/src/main/java/org/eclipse/che/ide/editor/orion/client/OrionEditorWidget.java (hunk 1 above)
Comment (evidolob) [reply ide/che-core-orion-editor/src/main/java/org/eclipse/che/ide/editor/orion/client/OrionEditorWidget.java:198]: I disagree, with extra `((`  its hard to read which method result is cast

**Comment by tsmaeder** [comment]
Is the fix I did here: https://github.com/eclipse/che/pull/5442/commits/c283fbe4b939e46c80087a4cc0859283b41bbe6f#diff-636bde1180f7be669df55d396a611267 not enough to fix the problem?

**Comment by evidolob** [comment]
No, the problem is in `showProblems` method, in Orion 14 Spell checker uses same method to show spell errors which override annotations provided by LS or Java plugin

**Comment by tsmaeder** [comment]
@evidolob then why don't we stop using "showErrors" altogether? Wouldn't binding the Che annotation model to the Orion annotation model make more sense?

**Comment by evidolob** [comment]
@tsmaeder In this pull we stop. I mean Orion js method, not java method

**Review by vparfonov — APPROVED** [review]

**Review by vparfonov — APPROVED** [review]

**Pull request merged**